ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Node/Serialisation.hs

```diff
@@ -57,37 +56,41 @@ instance ShelleyCompatible proto era => SerialiseDiskConstraints (ShelleyBlock p
 
 instance ShelleyCompatible proto era => EncodeDisk (ShelleyBlock proto era) (ShelleyBlock proto era) where
   encodeDisk _ = encodeShelleyBlock
-instance ShelleyCompatible proto era => DecodeDisk (ShelleyBlock proto era) (Lazy.ByteString -> ShelleyBlock proto era) where
+instance ShelleyCompatible proto era => DecodeDisk (ShelleyBlock proto era) (ShelleyBlock proto era) where
   decodeDisk _ = decodeShelleyBlock
 
 instance ShelleyCompatible proto era => EncodeDisk (ShelleyBlock proto era) (Header (ShelleyBlock proto era)) where
   encodeDisk _ = encodeShelleyHeader
-instance ShelleyCompatible proto era => DecodeDisk (ShelleyBlock proto era) (Lazy.ByteString -> Header (ShelleyBlock proto era)) where
+instance ShelleyCompatible proto era => DecodeDisk (ShelleyBlock proto era) (Header (ShelleyBlock proto era)) where
   decodeDisk _ = decodeShelleyHeader
 
 instance ShelleyCompatible proto era => EncodeDisk (ShelleyBlock proto era) (LedgerState (ShelleyBlock proto era)) where
   encodeDisk _ = encodeShelleyLedgerState
 instance ShelleyCompatible proto era => DecodeDisk (ShelleyBlock proto era) (LedgerState (ShelleyBlock proto era)) where
-  decodeDisk _ = decodeShelleyLedgerState
+  -- ignore the bytestring passed in as it will be empty because the
+  -- LedgerState does not need to check anything on it.
```

Comment:
Could consider creating a function:
```
noNeedOriginalBytes :: (forall s. Decoder s a) -> ByteString -> (forall s. Decoder s a)
noNeedOriginalBytes f _ = f
```
so that it is clear that we expect the decoder to not need the bytestring.